### DISCUSSION TITLE
StreamLoad 在 1.2.1-rc01 和 2.0.0-beta 上的 json解析不一致

### DISCUSSION THREAD

**Discussion by csliubo** [apache/doris#21764] [A - General / Q&A] [answered]
我们有一个应用是通过 flink cdc 把 MySQL 数据同步到 doris。 我们最近把doris升级到 2.0.0-beta，某些有json字段的表同步就出错了。 

出错提示是：
`Reason: column(text) values is null while columns is not nullable. src line...`

对应的数据类似于：
`{
    "id": "1571742079333729121",
    "parent_id": null,
    "code": "CUSTOMER_MEMBER_CONFIG_LEVEL",
    "val": "6",
    "text": "[{\"text\": \"............\", \"language\": \"zh-cn\"}]",
    "state": "1",
    "sort": "6",
    "create_time": "2022-12-12 15:51:39",
    "creator": "1571742079333728257",
    "modify_time": "2023-06-26 17:42:50",
    "modifier": null,
    "is_del": "0",
    "version": "0",
    "__DORIS_DELETE_SIGN__": "0"
}`
实际上这个 text 字段是有值的，是被转义之后的 json string。text 字段是在mysql里是json类型，debezium 序列化之后这个值会被转义。 之前在 1.2.1 版本上是可以处理这种转义后的 json string的，但在2.0.0-beta上出现问题。 

2.0.0-beta 上的这种处理是属于特性还是 bug？ 如果是特性我们就自己来处理这个转义问题了....

**Comment by LemonLiTree** [comment, marked as answer]
I reproduced it, it should be a bug, let's fix it

**Comment by eldenmoon** [comment]
已修复

**Reply by eldenmoon** [reply]
https://github.com/apache/doris/pull/21866
